File: dynamic_programming/746. Min Cost Climbing Stairs/minCostClimbingStairs.js
/**
 * @param {number[]} cost
 * @return {number}
 */
var minCostClimbingStairs = function(cost) {
    const n = cost.length;
    for (let i = 2; i < cost.length; i++) {
        cost[i] += Math.min(cost[i-1], cost[i-2]);
    }
    return Math.min(cost[n-2], cost[n-1]);
};

File: dynamic_programming/746. Min Cost Climbing Stairs/minCostClimbingStairs.py
class Solution(object):
    def minCostClimbingStairs(self, cost):
        """
        :type cost: List[int]
        :rtype: int
        """
        # Update the values of the list: each step contains the minimum cost to reach it
        for i in range(2, len(cost)):
            # We either came from the prev step or the one before it
            # We choose the one having the minimum sum cost
            cost[i] += min(cost[i-1], cost[i-2])
        
        # We reach the end (len(cost)), by either taking 2 steps from -2, or 1 step from last step -1
        return min(cost[-1], cost[-2])